subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -394,7 +395,11 @@ static void firmware_transfer(struct k_work *work)
 	char *server_addr;
 
 #if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_SUPPORT)
-	server_addr = CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_ADDR;
+	if (cfg_proxy_uri == NULL) {
```

Comment:
If we assign cfg_proxy_uri up above to CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_ADDR, then we can remove this check, and default to `server_addr = cfg_proxy_uri`